src/blocks/collapsible-control/frontend.js

```diff
@@ -0,0 +1,224 @@
+bu_blocks.collapsibleControl = ( function() {
+
+	// Store all Control blocks
+	var collapsibleControlBlocks = [];
+	var allCollapsibleBlocks = [];
+	var allBlocksOpen = false;
+	var collapsibleOpenClass = 'is-open';
+
+	/**
+	 * Open or close a group of collapsible blocks
+	 *
+	 * @param array collapsible blocks
+	 * @param bool true to open set of collapsible blocks, false to close
+	 */
+	var controlCollapsibleBlocks = function( collapsibleBlocks, open = true ) {
+
+		collapsibleBlocks.forEach( function( collapsible, i ) {
+
+			const { container, toggle, panel } = collapsible;
+
+			if ( open ) {
+
+				container.classList.add( collapsibleOpenClass );
+				toggle.setAttribute( 'aria-expanded', true );
+				panel.setAttribute( 'aria-hidden', false );
+
+			} else {
+
+				container.classList.remove( collapsibleOpenClass );
+				toggle.setAttribute( 'aria-expanded', false );
+				panel.setAttribute( 'aria-hidden', true );
+
+			}
+
+		} );
+
+	}
+
+	/**
+	 * Toggle all Collapsible blocks
+	 */
+	var toggleAll = function( control ) {
+
+		if ( 0 === allCollapsibleBlocks.length ) {
+			return;
+		}
+
+		controlCollapsibleBlocks( allCollapsibleBlocks, !allBlocksOpen );
+
+		allBlocksOpen = ( allBlocksOpen ) ? false : true;
+
+	};
+
+	/**
+	 * Toggle Collapsible blocks in control's group
+	 */
+	var toggleGroup = function( control ) {
+
+		const { groupIsOpen, collapsibleBlocks } = control;
+
+		controlCollapsibleBlocks( collapsibleBlocks, !groupIsOpen );
+
+		control.groupIsOpen = ( groupIsOpen ) ? false : true;
+
+	};
+
+	/**
+	 * Find all Collapsible blocks on a page
+	 */
+	var findAllCollapsibleBlocks = function() {
+
+		var containers = document.querySelectorAll( '.wp-block-editorial-collapsible' );
+
+		// Don't coninue if no Collapsible blocks exist
+		if ( containers.length === 0 ) {
+			return;
+		}
+
+		containers.forEach( function( element, i ) {
+
+			var block = {};
+
+			block.container = element;
+			block.toggle = element.querySelector( '.bu-block-collapsible-toggle' );
+			block.panel = element.querySelector( '.bu-block-collapsible-content' );
+
+			allCollapsibleBlocks.push( block );
+
+		} );
+
+	};
+
+	/**
+	 * Return all Collapsible blocks in the group with a Control
+	 *
+	 * @param object control
+	 *
+	 * @return array list of all collapsible blocks in group
+	 */
+	var getGroupCollapsibleBlocks = function( control ) {
+
+		var blocks = [];
+
+		var group = control.closest( '.wp-block-group' );
+		var containers = group.querySelectorAll( '.wp-block-editorial-collapsible' );
```

Comment:
this querySelectorAll is throwing an error in my sandbox. 

```
Cannot read property 'querySelectorAll' of null
    at getGroupCollapsibleBlocks (bu-blocks-frontend.js?ver=1610573994:617)
    at bu-blocks-frontend.js?ver=1610573994:661
    at NodeList.forEach (<anonymous>)
    at findElements (bu-blocks-frontend.js?ver=1610573994:650)
    at collapsibleControlInit (bu-blocks-frontend.js?ver=1610573994:723)
    at HTMLDocument.<anonymous> (bu-blocks-frontend.js?ver=1610573994:732)
```

Reply:
Updated. let me know if all resolved. thanks

Reply:
I don't see the error anymore so 👍